packages/server-wallet/src/handlers/single-app-updater.ts

```diff
@@ -0,0 +1,51 @@
+import {WirePayload} from '../type-aliases';
+import {WalletResponse} from '../wallet/wallet-response';
+import {Store} from '../wallet/store';
+
+/**
+ * For making a single update to a running application channel
+ * */
+export class SingleAppUpdater {
+  private store: Store;
+
+  public static create(store: Store): SingleAppUpdater {
+    return new SingleAppUpdater(store);
+  }
+
+  private constructor(store: Store) {
+    this.store = store;
+  }
+
+  /**
+   * For pushing a message containing a single update to a running application channel
+   */
+  async update(wirePayload: WirePayload, response: WalletResponse): Promise<void> {
+    // check that we have exactly one signedState and nothing else
+    if (wirePayload.signedStates?.length !== 1)
+      throw new Error(`The payload sent to pushUpdate must contain exactly 1 signedState.`);
+    if (wirePayload.objectives)
+      throw new Error(`The payload sent to pushUpdate must not contain objectives.`);
+    if (wirePayload.requests)
+      throw new Error(`The payload sent to pushUpdate must not contain requests.`);
+
+    const wireState = wirePayload.signedStates[0];
+    const channelId = wireState.channelId;
+
+    await this.store.transaction(async tx => {
+      const channel = await this.store.addSignedState(channelId, wireState, tx);
+
+      // check channel is an application channel. If not, transaction will rollback when promise rejects.
+      if (!channel.isAppChannel)
+        throw new Error(`The update sent to pushUpdate must be for an application channel`);
+
+      // check that the channel is running. If not, transaction will rollback when promise rejects
+      // note that this actually checks that the channel is still running _after_ the update has
+      // been applied, as this is the case where no cranking is required
+      if (!channel.isRunning)
+        throw new Error(`The update sent to pushUpdate must be for a running channel`);
```

Comment:
Thinking about @snario's scenario:  this places a requirement on the app to understand what _not_ to send through this method. I think we should alter the docstring on the API to reflect this more fully, i.e. explicitly mention to not send `isFinal` states?

Reply:
This was my opposition to this method when speaking with Tom last week. I'd prefer if we only had `pushMessage` and defaulted `doAnotherCrank` to be `true` unless this `isRunning` check is `true`

Reply:
I agree with this. I think it's worth merging this (to get the short-term benefit, as it seems to fit well with the-graph's use-case) but scheduling work to make the logic part of pushMessage later.